Documentation/Contributors/CodingGuide/README.md

```diff
@@ -0,0 +1,662 @@
+
+:speedboat: The speedboat indicates a performance tip.
+
+To some extent, this guide can be summarized as _make new code similar to existing code_.
+
+_TODO: TOC_
+
+## Naming
+
+* Directory names are `PascalCase`, e.g., `Source/Scene`.
+* Constructor functions are `PascalCase`, e.g., `Cartesian3`.
+* Functions are `camelCase`, e.g., `defaultValue()`, `Cartesian3.equalsEpsilon()`.
+* Files end in `.js` and have the same name as the JavaScript identifier, e.g., `Cartesian3.js` and `defaultValue.js`.
+* Variables, including class properties, are `camelCase`, e.g.,
+```javascript
+this.minimumPixelSize = 1.0; // Class property
+
+var bufferViews = gltf.bufferViews; // Local variable
+```
+* Private (by convention) members start with an underscore, e.g.,
+```javascript
+this._canvas = canvas;
+```
+* Constants are in uppercase with underscores, e.g.,
+```javascript
+Cartesian3.UNIT_X = freezeObject(new Cartesian3(1.0, 0.0, 0.0));
+```
+* Avoid abbreviations in public identifiers unless the full name is prohibitively cumbersome and has a widely accepted abbreviation, e.g., 
+```javascript
+Cartesian3.maximumComponent() // not Cartesian3.maxComponent()
+
+Ellipsoid.WGS84 // not Ellipsoid.WORLD_GEODETIC_SYSTEM_1984
+```
+* Prefer short and descriptive names for local variables, e.g., if a function has only one length variable,
+```javascript
+var primitivesLength = primitives.length;
+```
+is better written as
+```javascript
+var length = primitives.length;
+```
+* When accessing an outer-scope's `this` in a closure, name the variable `that`, e.g.,
+```javascript
+var that = this;
+this._showTouch = createCommand(function() {
+    that._touch = true;
+});
+```
+
+A few more naming conventions are introduced below along with their design pattern, e.g., [`options` parameters](#options-parameters), [`result` parameters and scratch variables](#result-parameters-and-scratch-variables), and [`from` constructors](#from-constructors).
+
+## Formatting
+
+In general, format new code the same as the existing code.
+
+* Use four spaces for indentation.  Do not use tab characters.
+* Do not include trailing whitespace.
+* Put `{` on the same line as the previous statement:
+```javascript
+var defaultValue = function(a, b) {
+   // ...
+};
+
+if (!defined(result)) {
+   // ...
+}
+```
+* Use parenthesis judiciously, e.g.,
+```javascript
+var foo = x > 0.0 && y !== 0.0;
+```
+is better written as
+```javascript
+var foo = (x > 0.0) && (y !== 0.0);
+```
+* Use vertical whitespace to separate functions and to group related statements within a function, e.g.,
+```javascript
+var Model = function(options) {
+    // ...
+    this._allowPicking = defaultValue(options.allowPicking, true);
+
+    this._ready = false;
+    this._readyPromise = when.defer();
+    // ...
+};
+```
+* Use single quotes, `'`, instead of double quotes, `"`.  `"use strict"` is an exception and should use double quotes.
```

Comment:
I've actually been tempted to open a pull request to fix this, there's never be a good reason for use strict to be in double quotes and it's one of the oddest things about our code base.  I know you are just reporting what's already in the code here, just bringing this up in case anyone has an objection.

Also, you might want to mention that we use double quotes in HTML.